homeassistant/components/hive/sensor.py

```diff
@@ -45,6 +45,49 @@
         device_class=SensorDeviceClass.TEMPERATURE,
         entity_category=EntityCategory.DIAGNOSTIC,
     ),
+    SensorEntityDescription(
+        key="Heating_Current_Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        icon="mdi:thermometer",
+    ),
+    SensorEntityDescription(
+        key="Heating_Target_Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        icon="mdi:thermometer",
+    ),
+    SensorEntityDescription(
+        key="Heating_State",
+        icon="mdi:radiator",
+    ),
+    SensorEntityDescription(
+        key="Heating_Mode",
+        icon="mdi:radiator",
+    ),
+    SensorEntityDescription(
+        key="Heating_Boost",
+        icon="mdi:radiator",
+    ),
+    SensorEntityDescription(
+        key="Hotwater_State",
+        icon="mdi:hand-water",
+    ),
+    SensorEntityDescription(
+        key="Hotwater_Mode",
+        icon="mdi:hand-water",
+    ),
+    SensorEntityDescription(
+        key="Hotwater_Boost",
+        icon="mdi:hand-water",
+    ),
+    SensorEntityDescription(
+        key="Availability",
+        icon="mdi:check-circle"
+    ),
```

Comment:
Can you give examples of what the state would look like?

Reply:
Does this help:

![image](https://github.com/user-attachments/assets/25f23a6f-cd67-49dc-8bd3-87b44c041870)

Reply:
If there are only 2 states (on/off), they should be binary sensors.

I know we had this problem in the previous PR, but instead of trying to get into the same discussion, can we try getting into the constructive discussion on how we can get the integration on a point where we can add those?

Feel free to message me on Discord? I'm happy to help out and support where needed.

Reply:
The reason I chose a generic sensor over a binary sensor is as follows:

The data returned from Hive contains a number of elements, which include `sensor` data and `binary_sensor` data.  Hive clearly makes a distinction between the two.

All the sensors that I added appear in the `sensor` data element (and not `binary_sensor`).  Therefore, I suspect that the values may not be binary.  Even if they are binary at the moment, there's a risk that this could change in the future.

Reply:
But we add entities to what they are today, and not on what they could be in the future. If in the future it returns, idk 5 different modes instead of "ON", we have to make a breaking change and adapt the integration

Reply:
OK.  I've converted the binary sensor candidates, but they always display 'On' regardless of the true state of the underlying entity.  (I haven't pushed my changes.)

Reply:
That sounds incorrect. I was also quite surprised how the native value is acquired from the device in the sensor, can you maybe explain a bit how Hive works?

Reply:
I really have no idea on how Hive works internally.  I'm just a simple home user that's trying to get some missing functionality working in the HA Hive integration.

Reply:
Do you have discord by any chance?

Reply:
yes, just joined.

Reply:
I see 2 ribbals in the HA discord, can you send me a message?